### PR TITLE
[security] Inline Js Fixes, Fixes File Browser Issues and other related problems

### PR DESCRIPTION
**Pull request by tabraiz12**
## What changes were proposed in this pull request?

* Never Intended to create such a long PR.  PR consists of existing js in a new file, named as {filename}-inline. 
* Few variables needed for frontend has been exposed via a application/json tag. If given a choice would be using api calls. But that become more bigger change. hence sticked with the current approach.
* Have Blacklisted apps in appmanager.py. Those apps will not work in case when csp nonce is enabled.
* Other good point to take notice of is in onePageViewModel.js, there are two code paths. where inline js gets inserted and other where js file is loaded. 

## How was this patch tested?

- Working on creating a sheet, with details of testing done.

Please review [Hue Contributing Guide](https://github.com/cloudera/hue/blob/master/CONTRIBUTING.md) before opening a pull request.

### CODE REVIEW

File: desktop/core/src/desktop/js/onePageViewModel.js [hunk 1]
```diff
@@ -849,6 +927,9 @@ class OnePageViewModel {
       return filePathParam.replaceAll('%25', '%');
     };
 
+    // Register the nonce check middleware globally
+    // page('*', nonceCheckMiddleware);
+
```
Comment (JohanAhlen) [inline desktop/core/src/desktop/js/onePageViewModel.js:932]: Nit: remove?

**Review by JohanAhlen — APPROVED** [review]
Massive change! Nice work!

I noticed we dropped the I18n in the transition to js files. When you have some time you could use `window.I18n` and register the strings in `global_js_constants.mako`, it's not urgent but should be done at some point.

File: desktop/core/src/desktop/middleware.py [hunk 2]
```diff
@@ -884,6 +884,8 @@ def process_response(self, request, response):
     if self.secure_content_security_policy and 'Content-Security-Policy' not in response:
       response["Content-Security-Policy"] = self.secure_content_security_policy
 
+
+
```
Comment (Harshg999) [inline desktop/core/src/desktop/middleware.py:888]: Nit: To remove?

File: desktop/libs/indexer/src/indexer/views.py [hunk 3]
```diff
@@ -18,9 +18,8 @@
 import sys
 import json
 import logging
-
+import beeswax
```
Comment (Harshg999) [inline desktop/libs/indexer/src/indexer/views.py:21]: Is this used?

File: desktop/core/src/desktop/appmanager.py [hunk 4]
```diff
@@ -261,6 +261,9 @@ def load_apps(app_blacklist):
   DESKTOP_APPS = []
 
   for sdk_app in pkg_resources.iter_entry_points("desktop.sdk.application"):
+    
+    if desktop.conf.CSP_NONCE.get():
+      app_blacklist.extend(['oozie', 'jobbrowser', 'rdbms', 'search', 'zookeeper'])
```
Comment (Harshg999) [inline desktop/core/src/desktop/appmanager.py:266]: Suggestion: Can you add a comment why we are extending the blacklist here for specific apps?

File: apps/useradmin/src/useradmin/views.py [hunk 5]
```diff
@@ -73,22 +73,29 @@ def is_ldap_setup():
...
       'users_json': json.dumps(list(User.objects.values_list('id', flat=True))),
-      'request': request,
       'is_embeddable': request.GET.get('is_embeddable', False),
       'is_ldap_setup': is_ldap_setup()
-  })
+  }
+
+  # options_json is used in the UI
+  data['options_json'] = json.dumps(data)
+  data['request'] = request
```
Comment (Harshg999) [inline apps/useradmin/src/useradmin/views.py:84]: Sending the whole request back? Is this used?

File: apps/filebrowser/src/filebrowser/views.py [hunk 6]
```diff
@@ -17,6 +17,9 @@
 
 import io
 import os
+import json
+from django.utils.html import mark_safe
+from django.urls import reverse
```
Comment (Harshg999) [inline apps/filebrowser/src/filebrowser/views.py:22]: Are these new imports used?

**Review by Harshg999 — APPROVED** [review]
Nice change! Added few review comments...

File: apps/useradmin/src/useradmin/views.py (hunk 5 above)
Comment (tabraiz12) [reply apps/useradmin/src/useradmin/views.py:84]: This is similar to existing code, and it is used in the the mako template. Its just that i have removed request from the options_json as it is not required in the UI.

**Pull request merged**